Testing improvements

murrant opened the pull request:
Please review.  Don't merge just yet.

murrant committed:
Let laravel name the indexes automatically.  This allows us to use sqlite.

murrant committed:
Use sqlite memory db for testing by default

murrant committed:
Tests fixes
Fix user creation
Fix some items with the user table migration
Change baseUrl port
Device generator (unfinished)

murrant committed:
Trim down the travis file, not initializing mysql should speed it up a bit
Removed exception for php 7.0, we should care about errors.

laf commented on tests/db/AddDefaultDataTest.php:
```diff
@@ -18,8 +18,8 @@ public function testAdddbSchemaVersion()
-                ['hostname' => 'localhost', 'sysName' => 'mysystem', 'version' => '1.1', 'hardware' => 'Intel x64', 'location' => 'Some place in the world', 'status' => 1, 'status_reason' => ''],
-                ['hostname' => 'remotehost', 'sysName' => 'someonessystem', 'version' => '2.2', 'hardware' => 'AMD x64', 'location' => 'Some other place in the world', 'status' => 0, 'status_reason' => 'icmp'],
+                ['hostname' => 'localhost', 'ip' => '127.0.0.1', 'sysName' => 'mysystem', 'version' => '1.1', 'hardware' => 'Intel x64', 'location' => 'Some place in the world', 'status' => 1, 'status_reason' => ''],
+                ['hostname' => 'remotehost', 'ip' => '127.0.0.2', 'sysName' => 'someonessystem', 'version' => '2.2', 'hardware' => 'AMD x64', 'location' => 'Some other place in the world', 'status' => 0, 'status_reason' => 'icmp'],
```
ip column is a binary one so this needs to use inet-aton conversion

laf commented:
Thanks @murrant. Does dropping the index name still create the tables correctly?

murrant commented:
@laf, yes, Laravel generates the index name.  The problem was that many indexes had the same name, which isn't allowed in sqlite.  I'm not sure there would be any reason to know the name of the index.

murrant committed:
Properly encode ip when using raw db queries.

laf commented:
Glad you got it working :)

The /about page is failing phpunit tests on this one.

murrant commented:
@laf Yeah, I can't figure out the about page test.  It fails on my system even without these changes.  Any ideas?

laf committed:
Added testing for notifications page + archive

laf committed:
Added seeding of data + updated test cases

laf committed:
Re-added tearDown

laf committed:
updated testing env

murrant committed:
Merge pull request #1 from laf/phpunit

Phpunit updates + database seeding

murrant committed:
Use models and aggregate functions for about data.
Fix class case for seeders.

murrant committed:
Use models to seed database.

murrant committed:
Rename database connections from sqlite_testing to testing_sqlite, testing_mysql.
Don't reset the testing database when using a memory database.

murrant committed:
Move the testing DB_DEFAULT variable from TestCase to phpunit.xml

murrant committed:
Support sqlite version on the about page.

murrant committed:
Clean up TestCase and use the built in database migrations
Only seed in tests that need it
Move device insert to the seeder
Use eloquent models in list devices test

murrant committed:
Update install

murrant committed:
Merge remote-tracking branch 'upstream/develop' into develop

murrant committed:
Merge branch 'develop' into phpunit
